### ISSUE TITLE
URLs with encoded characters not handled properly. 

### ISSUE THREAD

**Issue by computator** [cloudflare/kv-asset-handler#75]
If I have an asset with a filename `test file.jpg` the corresponding url is `/test%20file.jpg`, which is correct. However, `getAssetFromKV` tries to access it as `test%20file.jpg` which does not exist and fails.

the `URL` constructor converts all spaces not already encoded to `%20` automatically, so a simplistic solution would be running `decodeURI` here  after the url was already parsed with `URL`:
https://github.com/cloudflare/kv-asset-handler/blob/d4fa91fae1806e30270dcc5f39f581ea37362a59/src/index.ts#L101

**Comment by EverlastingBugstopper** [comment]
Hey @computator - this is actually an issue with the Workers runtime itself (very longstanding). In our [docs](https://developers.cloudflare.com/workers/reference/apis/standard/) we recommend polyfilling the URL class with a [better](https://npmjs.com/package/universal-url) [implementation](https://npmjs.com/package/whatwg-url) if you need to. Let me know if you need any help figuring that out 😄

**Comment by computator** [comment]
The runtime URL class may be nonconformant, but the percent encoding aspect of it is: https://url.spec.whatwg.org/#example-5434421b. If you test it in chrome or something you can see it returns the pathname percent encoded as well:
```
> new URL('http://google.com/test%20path').pathname
'test%20path'
```
As such, it is in fact the kv-asset-handler code that is not compensating for that encoding before loading files from the KV store. 😁

**Comment by EverlastingBugstopper** [comment]
ooohhh - very interesting. 

I've become so conditioned to respond to URL encoding issues with a recommendation to polyfill that I didn't actually read your report.

Your diagnosis seems correct and using `decodeURI` should fix 👍 

Happy to accept a PR or we can add this to our list of things to address.

**Comment by pmkemper** [comment]
Any update on this? 

When using filenames in non-latin scripts, like Cyrillic or Arabic, kv-asset-handler will also look for the encoded name instead of the decoded name.

Using `decodeURI` fixes this. But it is futile to use decodeURI in `mapRequestToAsset`, because `Request()` encodes the URL right back. So decodeURI needs to be run in kv-asset-handler itself.

**Comment by ericselin** [comment]
I think this is a little insane. If you have a CMS (which most sites probably do), and you allow editors to upload images, for example (which also, you probably do), you're bound to have some image filenames with a space in them. **And spaces in a filename does not work!**

This broke our production site yesterday. I would consider this package core to the Worker sites offering, so [Cloudflare] please don't hide behind the "yeah, just fix it yourself and open a PR if you care about this" argument. Personally I think that wrangler should encode the filenames, but I don't know your [Cloudflare] development plans.

**Comment by SukkaW** [comment]
I think the issue could be closed now. My PR #100 & #105 have fixed the issue.

**Comment by ispivey** [comment]
This will be fixed shortly in 0.0.11.
